Add Acl whitelist Accounts + Contracts (#217)
casper-ecosystem/cep-78-enhanced-nft: README.md

mode cannot be changed once the contract has been installed.
The mode is passed in as a `u8` value to `nft_holder_mode` runtime argument.

#### Minting

The minting mode governs the behavior of contract when minting new tokens. The minting modality provides two options:

1. `Installer`: This mode restricts the ability to mint new NFT tokens only to the installing account of the NFT contract.
2. `Public`: This mode allows any account to mint NFT tokens.

> 3. `Acl`: This mode allows whitelisted account/contract/contract from a package to mint NFT tokens.

Reviewer: ```suggestion
3. `Acl`: This mode allows whitelisted accounts, contracts or contracts from a package to mint NFT tokens.
```

Reviewer: Is Acl an acronym? If so, it should be written as ACL.

Author: Renamed to ACL thanks

> 3. `Acl`: This mode allows whitelisted account/contract/contract from a package to mint NFT tokens.

Author: Right now text is 

`This mode allows whitelisted accounts, contracts or contracts from a package to mint NFT tokens.` as a package hash does not mint directly.

| ----------- | --- |
| Installer   | 0   |
| Public      | 1   |
| Acl         | 2   |

#### WhitelistMode

> The `WhitelistMode` dictates if the Acl whitelist restricting access to the mint entrypoint can be updated. There are currently

Reviewer: ```suggestion
The `WhitelistMode` dictates if the Acl whitelist restricting access to the mint entry point can be updated. There are currently two options:
```

> two options:

Reviewer: ```suggestion
```

1. `Unlocked`: The Acl whitelist is unlocked and can be updated via the set variables endpoint.
2. `Locked`: The Acl whitelist is locked and cannot be updated further.

> If the `WhitelistMode` is set to `Locked` an Acl whitelist of entity keys must be provided on installation. This whitelist dictates which

Reviewer: ```suggestion
If the `WhitelistMode` is set to `Locked` an Acl whitelist of entity keys must be provided on installation. This whitelist dictates which entities can mint NFTs in the restricted `Acl` minting mode. These entities include `Accounts`, `Contracts` and `Contracts from a package`.
```

> entity either `Account` / `Contract` / `Contract from a package` are allowed to mint NFTs in the restricted `Acl` minting mode.

Reviewer: ```suggestion
```

These are showing up as separate lines, which is strange for a README and may cause formatting issues.

> This `WhitelistMode` is an optional installation parameter and will be set to unlocked if not passed. However, the whitelist mode itself

Reviewer: ```suggestion
This `WhitelistMode` is an optional installation parameter and will be set to unlocked if not passed. However, the whitelist mode itself cannot be changed once the contract has been installed. The mode is passed in as a `u8` value to `whitelist_mode` runtime argument.
```

> cannot be changed once the contract has been installed. The mode is passed in as a `u8` value to `whitelist_mode` runtime argument.

Reviewer: ```suggestion
```

- `"allow_minting"`: The `"allow_minting"` flag allows the installer of the contract to pause the minting of new NFTs. The `allow_minting` is a boolean toggle that allows minting when `true`. If not provided at install the toggle will default to `true`. This value can be changed by the installer by calling the `set_variables()` entrypoint.
- `"whitelist_mode"`: The [`WhitelistMode`](#whitelistmode) modality dictates whether the acl whitelist can be updated. This optional parameter will default to an unlocked whitelist that can be updated post installation. This parameter cannot be changed once the contract has been installed.
- `"holder_mode"`: The [`NFTHolderMode`](#nftholdermode) modality dictates which entities can hold NFTs. This is an optional parameter and will default to a mixed mode allowing either `Accounts` or `Contracts` to hold NFTs. This parameter cannot be changed once the contract has been installed.

> - `"acl_whitelist"`: The acl whitelist is a list of account / contract hashes / package that specifies which entity can call the `mint()` entrypoint to mint NFTs. This is an optional parameter which will default to an empty whitelist. This value can be changed via the `set_variables` post installation. If the whitelist mode is set to locked, a non-empty whitelist must be passed; else, installation of the contract will fail.

Reviewer: ```suggestion
- `"acl_whitelist"`: The acl whitelist is a list of account / contract hashes / packages that specifies which entity can call the `mint()` entrypoint to mint NFTs. This is an optional parameter which will default to an empty whitelist. This value can be changed via the `set_variables` post installation. If the whitelist mode is set to locked, a non-empty whitelist must be passed. If the whitelist mode is set to locked and you do not provide a non-empty whitelist, the contract will fail.
```

Author: @ACStoneCL thanks for the good watch
